src/common/ceph_argparse.cc

```diff
@@ -504,6 +504,9 @@ CephInitParameters ceph_argparse_early_args
     else if (ceph_argparse_witharg(args, i, &val, "--conf", "-c", (char*)NULL)) {
       *conf_file_list = val;
     }
+    else if (ceph_argparse_flag(args, i, "--no-config-file", (char*)NULL)) {
```

Comment:
another option is to call `conf.parse_argv()` twice. once before `conf.parse_config_files()`, so the `no_config_file` is populated. the other is after `conf.parse_env()`, so the options specified by command line take highest priority.